src/tests/ftest/aggregation/aggregation_basic.py

```diff
@@ -0,0 +1,126 @@
+"""
+
+import time
+from ior_test_base import IorTestBase
+
+class DaosAggregationBasic(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description:
+
+       Run IOR with same file option to verify the
+       aggregation happens in the background. Check for the
+       space usage before and after the aggregation to verify
+       the overwritten space is reclaimed.
+
+    :avocado: recursive
+    """
+
+    def test_basic_aggregation(self):
+        """Jira ID: DAOS-3451
+
+        Test Description:
+            Purpose of this test is to run ior using DFS api to write to
+            the DAOS and rerun it again with the same file option and
+            enable the aggregation and verify the aggregation reclaims
+            the overwritten space.
+
+        Use case:
+            Run ior with read, write, CheckWrite, CheckRead
+            Run ior with read, write, CheckWrite, CheckRead with same file
+            Enable aggregation and wait for 90 seconds to let the files
+            old enough to be marked for aggregation.
+            Wait for aggregation to complete.
+            Check the utilized capacity of the container/pool. It should be
+            roughly the same as after the initial run since aggregation has
+            reclaimed the overwritten capacity.
+
+        :avocado: tags=all,pr,hw,large,aggregate,daosio,aggregatebasic
+        """
+
+        # Create pool and container
+        self.update_ior_cmd_with_pool()
+
+        # Since the transfer size is 1M, the objects will be inserted
+        # directly into NVMe and hence storage_index = 1
+        storage_index = 1
+
+        pool_info = self.pool.pool.pool_query()
+        initial_free_space =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        # Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run ior with -k option to retain the file and not delete it
+        self.run_ior_with_pool()
+        pool_info = self.pool.pool.pool_query()
+        free_space_after_first_ior =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        space_used_by_ior = initial_free_space - free_space_after_first_ior
+
+        self.log.info("Space used by first ior = {}".format(
+            space_used_by_ior))
+        self.log.info("Free space after first ior = {}".format(
+            free_space_after_first_ior))
+        self.assertTrue(free_space_after_first_ior < initial_free_space,
+                        "IOR run was not successful.")
+
+        # Run ior the second time on the same pool and container, so another
+        # copy of the file is inserted in DAOS.
+        self.run_ior_with_pool(update=False)
+        pool_info = self.pool.pool.pool_query()
+        free_space_after_second_ior =\
+            pool_info.pi_space.ps_space.s_free[storage_index]
+
+        self.log.info("Free space after second ior = {}".format(
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments